src/lib/y2network/hwinfo.rb

```diff
@@ -17,14 +17,59 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
+require "yast"
+
 module Y2Network
-  # Physical interface hardware information
+  # Stores useful (from networking POV) items of hwinfo for an interface
+  # FIXME: decide whether it should read hwinfo (on demand or at once) for a network
+  # device and store only necessary info or just parse provided hash
   class Hwinfo
-    # @return [Hash] hwinfo
-    attr_accessor :data
+    attr_reader :hwinfo
+
+    def initialize(name:)
+      # FIXME: store only what's needed.
+      @hwinfo = load_hwinfo(name)
+    end
+
+    # Shortcuts for accessing hwinfo items
+    [
+      { name: "dev_name", default: "" },
+      { name: "mac", default: "" },
+      { name: "busid", default: "" },
+      { name: "link", default: false },
+      { name: "driver", default: "" },
+      { name: "drivers", default: [] },
+      { name: "requires", default: [] },
+      { name: "hotplug", default: false },
+      { name: "wl_auth_modes", default: "" },
+      { name: "wl_enc_modes", default: nil },
+      { name: "wl_channels", default: nil },
+      { name: "wl_bitrates", default: nil }
+    ].each do |hwinfo_item|
+      define_method hwinfo_item[:name].downcase do
+        @hwinfo ? @hwinfo.fetch(hwinfo_item[:name], hwinfo_item[:default]) : hwinfo_item[:default]
+      end
+    end
+    alias_method :name, :dev_name
+
+    def exists?
+      !@hwinfo.nil?
+    end
+
+    # Device type description
+    # FIXME: collision with alias for dev_name
+    def description
+      @hwinfo ? @hwinfo.fetch("name", "") : ""
```

Comment:
```suggestion
      name
```
or alias it (but I can never remember the correct syntax for that)

Reply:
i cannot ... I've already aliased dev_name to name

Reply:
`alias` and `alias_method` **add** a name, they don't move/destroy the old one. `alias_method :description, :dev_name` will just work. But don't trust me, just test it.

Reply:
I don't get it so I'll describe what i did and ask you if you can explain your requirement again

in hwinfo we have:
"dev_name" -> contains device name (eth0)
"name" -> textual device description like "Inter Ethernet Controller 82546"

I created a "name" alias for "dev_name" (```alias_method :name, :dev_name```)
and now what do you propose? Create an alias "description" for hwinfo's name item (alias_method :description, :name) ? I really don't know if it would behave correctly ...

Reply:
> in hwinfo we have:
> "dev_name" -> contains device name (eth0)
> "name" -> textual device description like "Inter Ethernet Controller 82546"

OK

> I created a "name" alias for "dev_name" (alias_method :name, :dev_name)

So now `name` no longer returns "Inter Ethernet Controller 82546", it returns `eth0`.

Anyway, your current implementation of `description` can be replaced by `alias_method :description, :name`.

Reply:
yes but is it correct? What if user uses ```hwinfo.name```? What is expected result in such case?

Reply:
First, let's forget about aliases, that's an implementation detail that has carried me away.

The important point IMHO is: For each element in `@hwinfo`, is its method name in `Hwinfo` the same? If not, there should be a comment with an explicit explanation. `Hwinfo` can have more than one name for the same thing, that's OK.

Reply:
| Hwinfo | `@hwinfo` |
|--|--|
| Hwinfo#dev_name | `@hwinfo["dev_name"]` |
| Hwinfo#name | `@hwinfo["dev_name"]` |
| Hwinfo#description | `@hwinfo["name"]` |